domain/src/test/kotlin/domain/actions/SwitchAllOffActionTest.kt

```diff
@@ -0,0 +1,50 @@
+package domain.actions
+
+import arrow.core.right
+import domain.Status
+import domain.asDeviceHost
+import domain.asDeviceId
+import domain.asThingId
+import domain.client.SwitchClient
+import domain.repository.DeviceRepository
+import domain.utils.aDeviceHost
+import domain.utils.aDeviceId
+import domain.utils.aThingId
+import domain.utils.aThingToDevice
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+
+class SwitchAllOffActionTest {
+
+    @MockK
+    private val deviceRepository: DeviceRepository = mockk()
+
+    @MockK
+    private val switchClient: SwitchClient = mockk()
+
+
+    private val switchAllOffAction = SwitchAllOffAction(deviceRepository, switchClient)
+
```

Comment:
MAJOR: missing test here, and also the functionality is wrong: what happens if one of the things is not switched off properly? Should it be fire and forget?

Reply:
At the moment is fire and forget for everything

Reply:
Here I should collect all the outcome, if any is not correct, it returns error -> 500 from API. This leads the UI to show a message saying "something went wrong, refresh the page please".